openedx/core/djangoapps/safe_sessions/middleware.py

```diff
@@ -457,92 +457,102 @@ def _verify_user(request, response, userid_in_session):
-                    '; '.join(f'{k}={v}' for (k, v) in session_hashes)
+                    'An unsafe user transition was found. It either needs to be fixed or exempted.\n' +
+                    '\n'.join(request.debug_user_changes)
                 )
-                for source_name, id_hash in session_hashes:
-                    set_custom_attribute(f'safe_sessions.session_id_hash.{source_name}', id_hash)
-                set_custom_attribute('safe_sessions.session_id_changed', session_id_changed)
-
-                if hasattr(request, 'debug_user_changes'):
-                    extra_logs.append(
-                        'An unsafe user transition was found. It either needs to be fixed or exempted.\n' +
-                        '\n'.join(request.debug_user_changes)
-                    )
-
-                if request_user_object_mismatch and not session_user_mismatch:
-                    log.warning(
-                        (
-                            "SafeCookieData user at initial request '{}' does not match user at response time: '{}' "
-                            "for request path '{}'.\n{}"
-                        ).format(  # pylint: disable=logging-format-interpolation
-                            request.safe_cookie_verified_user_id, request.user.id, request.path, '\n'.join(extra_logs)
-                        ),
-                    )
-                    set_custom_attribute("safe_sessions.user_mismatch", "request-response-mismatch")
-                elif session_user_mismatch and not request_user_object_mismatch:
-                    log.warning(
-                        (
-                            "SafeCookieData user at initial request '{}' does not match user in session: '{}' "
-                            "for request path '{}'.\n{}"
-                        ).format(  # pylint: disable=logging-format-interpolation
-                            request.safe_cookie_verified_user_id, userid_in_session, request.path, '\n'.join(extra_logs)
-                        ),
-                    )
-                    set_custom_attribute("safe_sessions.user_mismatch", "request-session-mismatch")
-                else:
-                    log.warning(
-                        (
-                            "SafeCookieData user at initial request '{}' matches neither user in session: '{}' "
-                            "nor user at response time: '{}' for request path '{}'.\n{}"
-                        ).format(  # pylint: disable=logging-format-interpolation
-                            request.safe_cookie_verified_user_id, userid_in_session, request.user.id, request.path,
-                            '\n'.join(extra_logs)
-                        ),
-                    )
-                    set_custom_attribute("safe_sessions.user_mismatch", "request-response-and-session-mismatch")
+        except BaseException as e:
+            log.error("SafeCookieData error while computing additional logs: %r", e)
```

Comment:
Nit/Question: Why not use?
```suggestion
            log.exception("SafeCookieData error while computing additional logs.")
```

Reply:
Why, not indeed? :-) Good to know about this option.